src-common/vcastServer.ts

```diff
@@ -196,84 +196,83 @@ export async function transmitCommand(
     statusText: "",
   };
 
-  await fetch(urlToUse)
-    .then(async (response) => {
-      // the server always returns an object with exitCode and data properties
-      const rawReturnData: any = await response.json();
+  try {
+    // Await axios post
+    const response = await axios.post(urlToUse, dataAsString, {
+      headers: {
+        "Content-Type": "text/plain",
+      },
+    });
+    const rawReturnData = response.data;
 
-      if (rawReturnData.exitCode == pythonErrorCodes.internalServerError) {
-        // the error message is a list of strings, so join with \n
-        transmitResponse.success = false;
-        transmitResponse.statusText = `Enviro server error: ${rawReturnData.data.error.join(
-          "\n"
-        )}`;
-        //
-        // if we get a test interface error, we report the error but do not
-        // terminate the server, since it might be a one off error.
-        // Maybe we will change this in the future.
-      } else if (
-        rawReturnData.exitCode == pythonErrorCodes.testInterfaceError
-      ) {
-        // the error message is a list of strings, so join with \n
-        transmitResponse.success = false;
-        // format the error message for readability, with new lines and indentation
-        transmitResponse.statusText = `Python interface error: \n   ${rawReturnData.data.text.join(
-          "\n   "
-        )}`;
-        //
-        // If we cannot start clicast, we shutdown the server because this error
-        // is unlikely to be cleared by trying again
-      } else if (
-        rawReturnData.exitCode == pythonErrorCodes.couldNotStartClicastInstance
-      ) {
-        transmitResponse.success = false;
-        transmitResponse.statusText = `Server could not start clicast instance`;
-        // fall back to non-server mode
-        setGLobalServerState(false);
-        // this callback will display an error message and update the test pane
-        await terminateServerProcessing(transmitResponse.statusText);
-        //
-        // else the command was successful so we return the data
-      } else {
-        transmitResponse.success = true;
-        transmitResponse.statusText = `Enviro server response status: ${response.statusText}`;
-        // the format of the data property is different based on the command
-        // so it is up to the caller to interpret it properly
-        transmitResponse.returnData = rawReturnData;
-        // there is always an exit code field but it is only used when
-        // executing tests or running clicast commands
-      }
-    })
+    // Handle different exit codes from the server
```

Comment:
response, not exit

Reply:
Changed in 008dd244ab1b3403497fe4f256c23ff43026b638.